### PR TITLE
NO-ISSUE: Ignoring private jobs in Prow

### PR DESCRIPTION
**Pull request by danmanor**
Fix scraper to take only public jobs from `Prow`

### CODE REVIEW

File: src/prowjobsscraper/scraper.py [hunk 1]
```diff
@@ -65,6 +65,8 @@ def _should_index_usage(
 
     @staticmethod
     def _is_assisted_job(j: prowjob.ProwJob) -> bool:
+        if j.spec.hidden:
```
Comment (rccrdpccl) [inline src/prowjobsscraper/scraper.py:68]: @osherdp do you know in what cases we can produce hidden jobs?

**Review by rccrdpccl — APPROVED** [review]

File: src/prowjobsscraper/scraper.py (hunk 1 above)
Comment (osherdp) [reply src/prowjobsscraper/scraper.py:68]: I think they enable having private jobs in repositories that are not exposed to all
For us it's not that interesting, although there's https://github.com/openshift/release/blob/a5d44ded681bcb6e7758b1c2b0801cf69feb11b9/ci-operator/config/openshift-priv/assisted-service/openshift-priv-assisted-service-master.yaml

Comment (rccrdpccl) [reply src/prowjobsscraper/scraper.py:68]: we do have a job there though: https://prow.ci.openshift.org/view/gs/origin-ci-private/logs/branch-ci-openshift-priv-assisted-service-release-4.14-images/1659126313992065024
Is this to be expected? I believe this did not happen before, so it must be something new

**Pull request merged**